pkg/controller/upgradeconfig/upgradeconfig_controller.go

```diff
@@ -180,29 +188,42 @@ func (r *ReconcileUpgradeConfig) Reconcile(request reconcile.Request) (reconcile
 			return reconcile.Result{}, nil
 		}
 	case upgradev1alpha1.UpgradePhaseUpgrading:
+		reqLogger.Info("Cluster detected as already upgrading.")
 		upgrader, err := r.clusterUpgraderBuilder.NewClient(r.client, metricsClient, instance.Spec.Type)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
-		reqLogger.Info("Cluster detected as already upgrading.")
-		err = upgrader.UpgradeCluster(instance, reqLogger)
-		if err != nil {
-			reqLogger.Error(err, "Failed to upgrade cluster")
-		}
+		return r.upgradeCluster(upgrader, instance, reqLogger)
 	case upgradev1alpha1.UpgradePhaseUpgraded:
 		reqLogger.Info("Cluster is already upgraded")
 		return reconcile.Result{}, nil
-	case upgradev1alpha1.UpgradePhaseFailed:
```

Comment:
This is not needed anymore?

Reply:
I don't see it used anywhere so i removed it. We return errors so they should be logged anyway